tools/ScenarioInOriginalGame/ScenarioCore.elm

```diff
@@ -53,3 +53,44 @@ setDirection direction playerId =
         (setStateComponentComment Dir playerId)
         (relativeAddressFor playerId Dir)
         direction
+
+
+type ScenarioCheckResult
+    = GoodScenario Scenario
+    | BadScenario String
```

Comment:
I’m not convinced this custom type provides any more clarity over `Result String Scenario`.

Reply:
You're right! I reached for `Either` and forgot about `Result`.

Reply:
Fixed in 4d73cf6.